hazelcast/src/main/java/com/hazelcast/quorum/QuorumConfig.java

```
@@ -0,0 +1,117 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.quorum;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.quorum.QuorumType.READ_WRITE;
+
+/**
+ * Contains the configuration for cluster quorum.
+ */
+public class QuorumConfig {
+
+    private String name;
+    private boolean enabled;
+    private int size;
+    private List<QuorumListenerConfig> listenerConfigs = new ArrayList<QuorumListenerConfig>();
+    private QuorumType type = READ_WRITE;
+
+    public QuorumConfig() {
+    }
+
+    public QuorumConfig(String name, boolean enabled) {
+        this.name = name;
+        this.enabled = enabled;
+    }
+
+    public QuorumConfig(String name, boolean enabled, int size) {
+        this.name = name;
+        this.enabled = enabled;
+        this.size = size;
+    }
+
+    public QuorumConfig(QuorumConfig quorumConfig) {
+        this.name = quorumConfig.name;
+        this.enabled = quorumConfig.enabled;
+        this.size = quorumConfig.size;
+        this.listenerConfigs = quorumConfig.listenerConfigs;
+        this.type = quorumConfig.type;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public QuorumConfig setName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public QuorumConfig setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        return this;
+    }
+
+    public int getSize() {
+        return size;
+    }
+
+    public QuorumConfig setSize(int size) {
```

No validation. This is a violation of the fail fast principle.